Enable macro unit tests

LeopoldGabelmann opened the pull request:
This is a:
- [ ] bug fix PR with no breaking changes
- [x] new functionality
- [ ] a breaking change

## Description & motivation
Within our team our yml for our dpt project got quite large. If one uses a lot of data tests, documents every column (even if it is centralized using docs) and then adds all the unit tests on top of it, it can be a lot of lines. Therefore, we were looking for a way to outsource all our unit tests inside of macros.
This has the additional benefit that it is very easy to run single a unit test while coding and controlling for possible errors. Furthermore, this makes it possible to run all data and unit tests apart from each other (e.g. in a CI/CD Pipeline) by selecting/excluding all unit tests via their folder path in tests.


## Checklist
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [x] Snowflake
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my macros (and models if applicable)

Commit:
Added all existing unit tests in macro form.

LeopoldGabelmann commented:
Hi @mjirv , 
another idea from me for your awesome package. I am very interested what you are going to think about that one, since it kind of changes your idea to have all tests at one place within the yml files. 
I tested the code on our snowflake instance with dbt version 1.5.0.

mjirv commented:
@LeopoldGabelmann thanks for submitting this! I think it's a great idea.

I'm taking a closer look, but would you mind adding something to the README about this option so people know it exists? Thanks for adding tests btw.

Commit:
Refactored the test blocks into macros blocks.
Replaced the original code inside the test blocks with the new macros.

Commit:
Adjusted the render logic for the the input mapping.
This has to be done for the values and the keys. This is necessary so that the unit tests can be executed as macros.

Commit:
Added explanations to the README on how to setup the tests as macros.

LeopoldGabelmann commented:
@mjirv - You're right, I should have added some explanation to the README on how to use this. I have done so now :)

LeopoldGabelmann commented:
@mjirv friendly reminder :)

santicapchase commented on macros/dmt_get_test_sql.sql:
```diff
@@ -123,13 +123,13 @@
-{% macro __set_rendered_keys(ns, keys) %}  
-    {% for k in keys %}
-        {% do ns.rendered_keys.update({k: render("{{ " + k + " }}")}) %}
+{% macro __set_rendered_mappings(ns, input_mapping) %}
```
FYI: This forces the ref in the `sql` files and the `yml` files to be EXACTLY the same
